src/Serializers/StatementListSerializer.php

```diff
@@ -69,12 +68,14 @@ public function serialize( $object ) {
 	private function getSerialized( StatementList $statementList ) {
 		$serialization = array();
 
-		$byPropertyIdGrouper = new ByPropertyIdGrouper( $statementList );
+		foreach ( $statementList->toArray() as $statement ) {
+			$idSerialization = $statement->getPropertyId()->getSerialization();
 
-		foreach ( $byPropertyIdGrouper->getPropertyIds() as $propertyId ) {
-			$serialization[$propertyId->getSerialization()] = $this->getSerializedStatements(
-				$byPropertyIdGrouper->getByPropertyId( $propertyId )
-			);
+			if ( !array_key_exists( $idSerialization, $serialization ) ) {
+				$serialization[$idSerialization] = array();
+			}
```

Comment:
Personally I think it's ok to omit this. As long as the thing is initialized as a (1-dimensional) array, PHP will happily add more dimensions when needed. This is tested, so changes in later PHP versions will instantly become visible.